packages/nestjs/src/test/ts/index.ts

```diff
@@ -1,9 +1,81 @@
-import {foo} from '../../main/ts'
+import {NestApplication} from '@nestjs/core'
+import {Controller, Get, HttpStatus} from '@nestjs/common'
+import {Test, TestingModule} from '@nestjs/testing'
+import request from 'supertest'
+import {SessionLdapProvider} from '@qiwi/ldap-common'
+import {LdapGuard, LdapRoles, LDAP_PROVIDER, UseLdapGuard} from '../../main/ts'
+import {testADProvider, testSessionProvider} from '../../test/stub/testLdap'
 
 describe('@qiwi/ldap-common', () => {
   describe('index', () => {
     it('properly exposes its inners', () => {
-      expect(foo).toBe('bar')
+      expect(LdapGuard).not.toBeUndefined()
     })
   })
+
+  describe('guard works correctly', () => {
+    let module: TestingModule
+    let app: NestApplication
+
+    const testLdapProvider = new SessionLdapProvider(
+      testADProvider,
+      testSessionProvider,
+      'test',
+    )
+
+    @Controller('cats')
```

Comment:
```typescript
@Controller('/')
    @UseLdapGuard()
    class CustomController {

      @Get('/private')
      @LdapRoles('Admin', 'User')
      getSmthPrivate(): string {
        return ['foo', 'bar']
      }

     @Get('/public')
      @LdapRoles('User')
      getSmthPublic(): string {
        return ['baz', 'qux']
      }
    }
}
```
token1 → roles: ['Admin', 'User'] should obtain data from both `/public` and `/private` endpoints
token2 → roles: ['User'] should have access to `/public` path only, and 403 for `/private`